### DISCUSSION TITLE
Will iced can support css in future?

### DISCUSSION THREAD

**Discussion by Anivie** [hecrj/iced#1366] [Q&A]
Just like javafx or qt, I think support for stylesheets is an important feature for gui frameworks, it allows us to reuse built-in components as much as possible without having to make new components when the functionality is the same

**Comment by basitmir7** [comment]
Rightly said ! No matter how many people hate CSS but it is a vital part of UI.

**Reply by BernardIgiri** [reply]
My person take is, CSS is great for web pages, but it is quite often overkill, or even counterproductive, for most applications. Most application development teams would not benefit from 10 different ways to center a box within a box. They don't need cascading styles, they are other ways to handle a z index, they don't need media queries, typography features (usually), floats, and clears, animations (plus there are many ways to handle animation), drop shadows, etc.. A desktop application can use components with custom rendering that is GPU accelerated and purpose built for specific tasks. Much of CSS's advanced features can be contained within custom components that are optimized for their specific purposes, thus eliminating the need for such CSS features. Additionally, in places where HTML needs to be rendered, an embedded browser is often a better solution than re-implementing all web technology into a GUI library.

**Reply by ngdangtu-vn** [reply]
I come from frontend background but I must say I don't like the idea of CSS in here (native GUI). CSS was designed for simplicity at first then grow up for more use cases. It is not suitable for more flexible, complicated and optimization cases. Moreover, any attempts to implement CSS in here only make developement experience way worse. Either dev team has to implement everything CSS spec has to match to popular documents on the internet (such as MDN) or make a new name for CSS and new document. 

I use anki and its CSS is horrible confused. I use Qt CSS and it is terrifying. I feels like my web dev exp is being 69 up. So I rather they don't do the same thing in here.

And I'm so glad they didn't.

**Comment by 13r0ck** [comment]
No. Iced has it's own styling system based on https://github.com/mdgriffith/elm-ui

Not all GUI toolkits have to be the same, and just because CSS is popular on the web doesn't mean it is the only, nor even best styling system. The Elm-UI inspired styling system works very well with the elm architecture that iced follows. It doesn't make sense to hack in support for another more complicated styling system, when the main styling system isn't even near completion!

And try learning something new, you might even like it! :grinning:

**Reply by richardanaya** [reply]
Elm-ui's last commit was 4 years ago. Css lives on.

**Reply by airstrike** [reply]
CSS lives on in the browser, which is not what `iced` targets, so I don't know how that's relevant here.

**Reply by richardanaya** [reply]
CSS is a language for styling, which is what this PR is about.  Whether it exists in browser in irrelevant.

**Comment by bungoboingo** [comment]
Iced will not support CSS. Time to move on, lads.

**Reply by richardanaya** [reply]
Are you a project owner?

**Reply by hecrj** [reply]
She is part of the core team. Can confirm; CSS support ain't happening.

Sorry, web devs.

**Reply by richardanaya** [reply]
Thanks

**Reply by ngdangtu-vn** [reply]
> She is part of the core team. Can confirm; CSS support ain't happening.
> 
> Sorry, web devs.

I'm web dev and  I'm glad this is the news :))

**Comment by bowenxuuu** [comment]
I don't think Iced needs CSS!
Another UI framework, Flutter, also doesn't support CSS, but it can still implement beautiful user interfaces, and that's enough!

**Comment by jomaro110** [comment]
What's needed in my opinion is a proper Figma to Iced plugin
